sweetest/src/main/java/com/mysugr/sweetest/framework/context/WorkflowTestContext.kt

```diff
@@ -1,56 +1,64 @@
 package com.mysugr.sweetest.framework.context
 
 import com.mysugr.sweetest.framework.flow.InitializationStep
-import com.mysugr.sweetest.framework.flow.InitializationStep.DONE
-import com.mysugr.sweetest.framework.flow.InitializationStep.INITIALIZE_DEPENDENCIES
-import com.mysugr.sweetest.framework.flow.InitializationStep.INITIALIZE_FRAMEWORK
 
 class WorkflowTestContext internal constructor(private val steps: StepsTestContext) {
 
-    var currentStep: InitializationStep = INITIALIZE_FRAMEWORK
-        private set
+    private var currentStep: InitializationStep = InitializationStep.INITIALIZE_FRAMEWORK
 
-    private val stepHandlers = InitializationStep.values().associate { it to mutableListOf<StepHandler>() }
+    private val supportedSubscriptionSteps = listOf(
+        InitializationStep.INITIALIZE_STEPS,
+        InitializationStep.INITIALIZE_DEPENDENCIES,
+        InitializationStep.SET_UP,
+        InitializationStep.RUNNING,
+        InitializationStep.TEAR_DOWN
+    )
 
-    private fun runStep(step: InitializationStep) {
-        currentStep = step
-        if (currentStep == INITIALIZE_DEPENDENCIES) {
-            onBeforeInitializeDependencies()
-        }
-        triggerHandler(step)
+    private val subscriptionHandlers =
+        supportedSubscriptionSteps.associate { it to mutableListOf<StepHandler>() }
```

Comment:
This could be simplified to:
```kotlin
private val subscriptionHandlers =
    InitializationStep.values().associate { it to mutableListOf<StepHandler>() }
```
Then we don't need the `supportedSubscriptionSteps` list.

Or is it intentional that not all Steps can be subscribed?

Reply:
yes that's intentional

```
requireNotNull(subscriptionHandlers[step]) { "Step \"$step\" isn't possible to be subscribed to" }
```